### PR TITLE
[Agency Dashboard][Refactor] CategoryOverview Refactor + Cleanup 

### PR DESCRIPTION
**Pull request by mxosman**
## Description of the change

Cleans up and refactors CategoryOverview component.

Playtesting: https://mahmoudtest---agency-dashboard-web-b47yvyxs3q-uc.a.run.app

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: agency-dashboard/src/CategoryOverview/CategoryOverview.tsx [hunk 1]
```diff
@@ -51,122 +46,82 @@ const categoryData: CategoryData = {
...
+    datapointsByMetric,
+    dimensionNamesByMetricAndDisaggregation,
+    loading,
+    downloadMetricsData,
+    getMetricsWithDataByCategory,
+  } = agencyDataStore;
+  const categoryKey = visibleCategoriesMetadata[category]?.key;
+  const metricsWithData = categoryKey
+    ? getMetricsWithDataByCategory(categoryKey)
+    : undefined;
```
Comment (mxosman) [inline agency-dashboard/src/CategoryOverview/CategoryOverview.tsx:68]: Should be the same exact list as in AgencyOverview. `AgencyOverview` uses `getMetricsWithDataByCategoryByCurrentSystem` which also uses `getMetricsWithDataByCategory` under the hood.

File: agency-dashboard/src/CategoryOverview/CategoryOverview.tsx [hunk 2]
```diff
@@ -51,122 +46,82 @@ const categoryData: CategoryData = {
...
-    dataRangeFilter,
+    datapointsByMetric,
+    dimensionNamesByMetricAndDisaggregation,
   });
 
   const { getBarChartData } = useBarChart({
-    getCurrentChartTimeRange,
-    datapointsByMetric: agencyDataStore.datapointsByMetric,
+    getDataVizTimeRange: getDataVizTimeRangeByMetricFrequency(dataRangeFilter),
+    datapointsByMetric,
```
Comment (mxosman) [inline agency-dashboard/src/CategoryOverview/CategoryOverview.tsx:82]: Haven't dug too deep into these hooks (that's up next) - but mostly cleaning up things that were unused and plugging in a centralized function to get the dataviz time range

File: agency-dashboard/src/utils/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      message: "Link copied",
+      color: "blue",
+    });
+  } catch {
+    showToast({
+      message: "Error copying link",
+      color: "blue",
+    });
+  }
+};
```
Comment (mxosman) [inline agency-dashboard/src/utils/helpers.ts:33]: Moved this out of the component

File: agency-dashboard/public/manifest.json [hunk 4]
```diff
@@ -1,21 +1,16 @@
...
-      "type": "image/png",
-      "sizes": "192x192"
-    },
-    {
-      "src": "logo512.png",
-      "type": "image/png",
-      "sizes": "512x512"
+      "src": "/android-chrome-512x512.png",
+      "sizes": "512x512",
+      "type": "image/png"
```
Comment (mxosman) [inline agency-dashboard/public/manifest.json:13]: [Side] Updating our favicon:
Screenshot 2023-08-25 at 1 10 30 PM to Screenshot 2023-08-25 at 1 14 12 PM

**Comment by mxosman** [comment]
This is the last one for now, @lilidworkin! No crashes and the dates seem to sync up nicely - but please let me know if you run into anything strange if you test.

File: agency-dashboard/public/manifest.json (hunk 4 above)
Comment (lilidworkin) [reply agency-dashboard/public/manifest.json:13]: Omg nice good catch!!

File: agency-dashboard/src/CategoryOverview/CategoryOverview.tsx (hunk 1 above)
Comment (lilidworkin) [reply agency-dashboard/src/CategoryOverview/CategoryOverview.tsx:68]: Awesome

**Review by lilidworkin — APPROVED** [review]
Great!!

**Comment by mxosman** [comment]
Thank you so much!

**Pull request merged**